proposals/0000-unstable.rst

```diff
@@ -0,0 +1,306 @@
+``-XUnstable``
+==============
+
+.. author:: Richard Eisenberg
+.. date-accepted::
+.. ticket-url::
+.. implemented::
+.. highlight:: haskell
+.. header:: This proposal is `discussed at this pull request <https://github.com/ghc-proposals/ghc-proposals/pull/524>`_.
+.. sectnum::
+.. contents::
+
+There are frequently new features in GHC that the GHC development team know are subject
+to change. Though we try to advertise features as unstable in e.g. the user manual and
+in release notes, we recognize it is impractical to expect every GHC user to read these
+resources.
+
+This proposal introduces a new extension ``-XUnstable``. Unstable features
+will be available only when ``-XUnstable`` is specified. Unstable features might
+change without notice, without a migration plan, and without a proposal, though
+notice of any changes would appear in release notes. In addition, we might choose
+to use this proposal process about unstable features, if collecting community
+feedback would be helpful.
+
+Moving
+an unstable feature out from the unstable umbrella would require a proposal.
+
+This proposal marks importing any module from ``ghc-prim`` as requiring ``-XUnstable``,
+though we expect to expand the set of features affected by ``-XUnstable`` over time.
+
+This proposal also includes a separate extension, ``-XStable``, which prevents
+``-XUnstable`` from being used. A project's ``.cabal`` file might then specify
+``-XStable`` to prevent any of the project's modules from using ``-XUnstable``.
+There is no ``-XNoStable``, unlike most other extensions. This ``-XStable`` component
+is optional and may be removed from this proposal without affecting the
+utility of ``-XUnstable``.
```

Comment:
I’d leave this out. So far we have no precedence for “.cabal file gets to set boundaries for haskell files”, and it assumes a division of roles with different “privileges” between those editing the .cabal file and those editing source files that I am not sure is realistic. 

Using `-XUnstable` is just one of many many things a local programming policy might want to forbid (such as other extensions, unsafe features, use of emoji keywords…), and is better served by a custom hlint database maybe.